openpype/plugins/publish/extract_hierarchy_avalon.py

```diff
@@ -159,20 +168,43 @@ def unarchive_entity(self, entity, data):
             {"_id": entity["_id"]},
             new_entity
         )
+
         return new_entity
 
-    def create_avalon_asset(self, project_name, name, data):
-        item = {
+    def create_avalon_asset(self, name, data):
+        asset_doc = {
             "schema": "openpype:asset-3.0",
             "name": name,
             "parent": self.project["_id"],
             "type": "asset",
             "data": data
         }
-        self.log.debug("Creating asset: {}".format(item))
-        entity_id = legacy_io.insert_one(item).inserted_id
+        self.log.debug("Creating asset: {}".format(asset_doc))
+        asset_doc["_id"] = legacy_io.insert_one(asset_doc).inserted_id
+
+        return asset_doc
+
+    def _set_avalon_data_to_relative_instances(self, context, asset_doc):
+        for instance in context:
+            asset_name = asset_doc["name"]
+            inst_asset_name = instance.data["asset"]
+
+            if asset_name == inst_asset_name:
+                instance.data["assetEntity"] = asset_doc
+
+                # get parenting data
+                parents = asset_doc["data"].get("parents") or list()
+
+                # equire only relative parent
+                if parents:
+                    parent_name = parents[-1]
```

Comment:
For assets without parents is parent name project name (`project_name` variable is missing).
```suggestion
                parent_name = project_name
                if parents:
                    parent_name = parents[-1]
```